### PR TITLE
xcore: add interface to pass external buffer to libxcam

### PR DESCRIPTION
**Pull request by zongwave**

### CODE REVIEW

File: xcore/Makefile.am [hunk 1]
```diff
@@ -51,6 +51,7 @@ xcam_sources = \
     interface/blender.cpp               \
     interface/geo_mapper.cpp            \
     interface/stitcher.cpp              \
+    once_map_video_buffer_priv.cpp      \
```
Comment (windyuan) [inline xcore/Makefile.am]: move once_map_video_buffer_priv.cpp together with other xcore cp files and keep interface/*.cpp separate.

File: xcore/once_map_video_buffer_priv.cpp [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+                              uint32_t size)
+{
+    VideoBufferInfo buf_info;
+    SmartPtr<OnceMapVideoBuffer> video_buffer;
+
+    XCAM_FAIL_RETURN (
+        ERROR, buf, NULL,
+        "external_buf_to_map_buf failed since buf is NULL");
+
+
```
Comment (windyuan) [inline xcore/once_map_video_buffer_priv.cpp]: one blank line is good enough

File: xcore/once_map_video_buffer_priv.cpp [hunk 3]
```diff
@@ -0,0 +1,101 @@
...
+    buf_info.init (format, width, height,
+                   aligned_width, aligned_height, size);
+    video_buffer = new OnceMapVideoBuffer (buf_info, buf);
+    XCAM_ASSERT (video_buffer.ptr ());
+    return video_buffer;
+}
+
```
Comment (windyuan) [inline xcore/once_map_video_buffer_priv.cpp]: one blank line is good enough

File: xcore/xcam_utils.h [hunk 4]
```diff
@@ -64,6 +64,15 @@ class VideoBuffer;
...
 
+SmartPtr<VideoBuffer>
+external_buf_to_once_map_buf (uint8_t* buf,
+                              uint32_t format,
+                              uint32_t width,
+                              uint32_t height,
+                              uint32_t aligned_width,
+                              uint32_t aligned_height,
+                              uint32_t size);
+
```
Comment (windyuan) [inline xcore/xcam_utils.h]: how about keep parameter style same as other functions when params too long. restart new line from first parameter.

	external_buf_to_once_map_buf (
		uint8_t* buf,  uint32_t format,
		uint32_t width, uint32_t height,
		uint32_t aligned_width, uint32_t aligned_height,
		uint32_t size);

**Comment by uartie** [comment]
Please ignore `intel/ci/ubuntu/zesty/pr` check status... there was a GitHub service interruption that caused the status to not get updated.

Instead, the real check is replaced by `intel/ci/ubuntu/zesty/pr/compile`.

**Comment by zongwave** [comment]
fixed, please help to review, thanks.

**Comment by zongwave** [comment]
commit 1: add interface to pass external buffer to libxcam
commit 2: correct vertex array object texture vertical coordinate into bottom to top order.

**Comment by windyuan** [comment]
merged. thanks.

**Pull request merged**